microfy.js

```diff
@@ -72,23 +73,30 @@ function build(
     })
   }
 
-  const service = new Proxy(
-    {
-      start: async () => {
+  const isNatsNeeded = subscriptions.length > 0
```

Comment:
This is not necessarily accurate, you could publish without having any subscriptions and still need to connect to NATS. Move this to a configuration param.